pkg/runtime.v2/core/trainingruntime.go

```diff
@@ -106,26 +103,23 @@ func (r *TrainingRuntime) buildObjects(
 		runtime.WithMLPolicy(mlPolicy),
 		runtime.WithPodGroupPolicy(podGroupPolicy),
 	}
-	for idx, rJob := range jobSetTemplateSpec.Spec.ReplicatedJobs {
-		replicas := jobSetTemplateSpec.Spec.ReplicatedJobs[idx].Replicas * ptr.Deref(rJob.Template.Spec.Completions, 1)
-		opts = append(opts, runtime.WithPodSpecReplicas(rJob.Name, replicas, rJob.Template.Spec.Template.Spec))
+	for _, rJob := range jobSetTemplateSpec.Spec.ReplicatedJobs {
+		// By default every ReplicatedJob has only 1 replica.
+		opts = append(opts, runtime.WithPodSpecReplicas(rJob.Name, 1, rJob.Template.Spec.Template.Spec))
 	}
-	info := runtime.NewInfo(&jobsetv1alpha2.JobSet{
-		TypeMeta: metav1.TypeMeta{
-			APIVersion: jobsetv1alpha2.SchemeGroupVersion.String(),
-			Kind:       "JobSet",
-		},
-		Spec: *jobSetTemplateSpec.Spec.DeepCopy(),
-	}, opts...)
-
-	if err := r.framework.RunEnforceMLPolicyPlugins(info); err != nil {
+
+	info := runtime.NewInfo(opts...)
+
+	if err := r.framework.RunEnforceMLPolicyPlugins(info, trainJob); err != nil {
 		return nil, err
 	}
-	err := r.framework.RunEnforcePodGroupPolicyPlugins(trainJob, info)
+
+	err := r.framework.RunEnforcePodGroupPolicyPlugins(info, trainJob)
```

Comment:
nit: we can combine the error statement in 1 line like above

Reply:
Good catch!